plugins/incremental-ingestion-backend/src/routes.ts

```diff
@@ -0,0 +1,230 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { errorHandler } from '@backstage/backend-common';
+import express from 'express';
+import Router from 'express-promise-router';
+import { Logger } from 'winston';
+import { IncrementalIngestionDatabaseManager } from './database/IncrementalIngestionDatabaseManager';
+
+/** @public */
+export const createIncrementalProviderRouter = async (
+  manager: IncrementalIngestionDatabaseManager,
+  logger: Logger,
+) => {
+  const router = Router();
+  router.use(express.json());
+
+  // Get the overall health of all incremental providers
+  router.get('/health', async (_, res) => {
+    const records = await manager.healthcheck();
+    const providers = records.map(record => record.provider_name);
+    const duplicates = [
+      ...new Set(providers.filter((e, i, a) => a.indexOf(e) !== i)),
+    ];
+
+    if (duplicates.length > 0) {
+      res.json({ healthy: false, duplicateIngestions: duplicates });
+    } else {
+      res.json({ healthy: true });
+    }
+  });
+
+  // Clean up and pause all providers
+  router.delete('/cleanup', async (_, res) => {
+    const result = await manager.cleanupProviders();
+    res.json(result);
+  });
+
+  // Get basic status of the provider
+  router.get('/:provider', async (req, res) => {
```

Comment:
You should really prefix these appropriately as a collection, so `'/providers/:provider'` here (and similarly below). Otherwise it gets really hard to evolve the API in the future. Besides, what happens otherwise when you add a provider whose name is `health`? :)

Reply:
Absolutely correct. Fixed!